xbmc/windowing/Resolution.cpp

```diff
@@ -144,6 +143,26 @@ void CResolutionUtils::FindResolutionFromWhitelist(float fps, int width, int hei
     }
   }
 
+  for (const auto& mode : indexList)
+  {
+    auto i = CDisplaySettings::GetInstance().GetResFromString(mode.asString());
+    const RESOLUTION_INFO info = CServiceBroker::GetWinSystem()->GetGfxContext().GetResInfo(i);
+
+    // allow resolutions that are exact in only one dimension and have the correct refresh rate
```

Comment:
This makes barely sense again. You would not want a 1920x800 video be played at 1280x800 resolution only cause the height matches ... it's wrong and will worsen current behaviour.

Reply:
Ok, I see your point. I didn't really intend to change the logic in this regard.
How about now? The only difference should be that truly exact matches (i.e. in both width _and_ height) are preferred over anything else, so the 4:3-case you mentioned should be displayed in the same resolution as before this PR.

Reply:
It makes basically no sense to Display a 1600 x800 video on a 1280x800 screen right? 

Because of the aspect ratio it would end up being displayed as 1280 x 600 or something. So even the 800 would not match.

I still think to break normal files while fixing a non native resolution in the whitelist is wrong to do. You cannot make it right for every user and especially not those that like shooting in their foot.

Reply:
My changes as they are right now should not affect the resolution choice in case of "a 1600 x800 video on a 1280x800 screen" and similar situations. It should **only** ensure that it accomodates what @vboginskeyin  proposed in https://github.com/xbmc/xbmc/issues/19037#issuecomment-756558351, i.e.

> To put it another way, would you agree that this logic is strictly better than what we have today:
> 
> 1. If: an exact resolution and refresh rate are available, choose that
> 2. Else: fall back to old behavior

Now, how to implement the above depends on the definition of "exact", of course. This PR is very strict in that regard, i.e. treating only pixel-perfect congruence in both width and height as "exact".